src/lib/y2network/connection_configs_collection.rb

```diff
@@ -61,6 +61,22 @@ def by_interface(interface_name)
       connection_configs.select { |c| c.interface == interface_name }
     end
 
+    # Returns connection with the given internal ID
+    #
+    # @param id [Integer] Internal ID
+    # @return [ConnectionConfig::Base] Connection config with the given ID
+    def by_id(id)
+      connection_configs.find { |c| c.id == id }
+    end
+
+    # Returns connections with any of the given internal IDs
+    #
+    # @param ids [Integer] Internal IDs
+    # @return [ConnectionConfig::Base] Connection config with the given IDs
```

Comment:
It should be in fact Array now or ConnectionsCollection if you use later suggestion.